### PR TITLE
Add match v2v3

### PR DESCRIPTION
**Pull request by ColinRCox**

### CODE REVIEW

File: pysiaf/aperture.py [hunk 1]
```diff
@@ -303,7 +303,7 @@ def complement(self):
             setattr(self, 'XIdlVert{:d}'.format(j), corners_Idl_x[j - 1])
             setattr(self, 'YIdlVert{:d}'.format(j), corners_Idl_y[j - 1])
 
-    def convert(self, X, Y, from_frame, to_frame):
+    def convert(self, X, Y, from_frame, to_frame) -> object:
```
Comment (Johannes-Sahlmann) [inline pysiaf/aperture.py]: Let's get rid of the check failures first. Please remove the `-> object` text (why is it here?). Then commit and push to the branch. That should allow the tests to be run.

**Comment by ColinRCox** [comment]
That weird —> object thing was put in by PyCharm. I found it in a few other places and removed them.
Will fix

**Comment by mperrin** [comment]
FYI that's a function annotation, specifying the return type. (not very usefully in this case). See https://www.python.org/dev/peps/pep-3107/

**Comment by Johannes-Sahlmann** [comment]
Thanks @mperrin for pointing this out.

File: pysiaf/aperture.py [hunk 2]
```diff
@@ -409,6 +409,27 @@ def get_polynomial_coefficients(self):
...
+        """Place a full set of coefficients into aperture
+        Input
+        : param sci2idlx - array of Sci2IdlX coefficients in JWST order
+        : param sci2idly - array of Sci2IdlY coefficients
+        : param idl2scix - array of Idl2SciX coefficients
+        : param idl2sciy - array of Idl2SciY coefficients
+        """
+
+        if self.Sci2IdlDeg is None:
+            return None
```
Comment (Johannes-Sahlmann) [inline pysiaf/aperture.py]: Instead of silently returning, please raise an error.

File: pysiaf/utils/tools.py [hunk 3]
```diff
@@ -443,3 +443,169 @@ def v3sciyangle_to_v3idlyangle(v3sciyangle):
...
+    if verbose:
+        print('Current Vref', aperture_2.V2Ref, aperture_2.V3Ref)
+        print('Shift to    ', V2Ref1, V3Ref1)
+
+    # Need to work in aperture 2  coordinate systems
+    aperName_1 = aperture_1.AperName
+    aperName_2 = aperture_2.AperName
+    detector_1 = aperName_1.split('_')[0]
+    detector_2 = aperName_2.split('_')[0]
+    print('Detector 1', detector_1, '  Detector 2', detector_2)
```
Comment (Johannes-Sahlmann) [inline pysiaf/utils/tools.py]: Please use `if verbose:`

File: pysiaf/utils/tools.py [hunk 4]
```diff
@@ -443,3 +443,169 @@ def v3sciyangle_to_v3idlyangle(v3sciyangle):
...
+    newC = CS
+    newD = DS
+
+    new_aperture_2 = copy.deepcopy(aperture_2)
+
+    # For NIRCam only, adjust angles
+    if instrument == 'NIRCAM':
+        newV3IdlYAngle = degrees(atan2(-AS[2], BS[2])) # Everything rotates by this amount
+        if abs(newV3IdlYAngle) > 90.0: newV3IdlYAngle = newV3IdlYAngle - copysign(180, newV3IdlYAngle)
+        print('New angle', newV3IdlYAngle)
```
Comment (Johannes-Sahlmann) [inline pysiaf/utils/tools.py]: `if verbose:`

File: pysiaf/tests/test_match_v2v3.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+from pysiaf import iando
+from pysiaf.utils import tools
+
+
```
Comment (Johannes-Sahlmann) [inline pysiaf/tests/test_match_v2v3.py]: Good point, that I had missed. Please define this as a function, e.g. 
`def test_match_v2v3_nircam():`

File: pysiaf/tests/test_match_v2v3.py [hunk 6]
```diff
@@ -1,29 +1,25 @@
 from pysiaf import iando
 from pysiaf.utils import tools
 
+def testit(instrument='NIRCam', apName_1 = 'NRCA1_FULL', apName_2 = 'NRCA5_FULL', verbose=False):
```
Comment (Johannes-Sahlmann) [inline pysiaf/tests/test_match_v2v3.py]: test function names should start with `test_` (incl. underscore) and be descriptive in what is being tested.

**Pull request merged**